backend/app/bayes_ab/models.py

```diff
@@ -218,33 +222,36 @@ async def get_all_bayes_ab_experiments(
 
 async def get_bayes_ab_experiment_by_id(
     experiment_id: int,
-    user_id: int,
+    user_id: int | None,
+    workspace_id: int,
     asession: AsyncSession,
 ) -> BayesianABDB | None:
     """
-    Get the A/B experiment by id.
+    Get the A/B experiment by id from a specific workspace.
     """
-    stmt = select(BayesianABDB).where(
-        and_(
-            BayesianABDB.user_id == user_id,
-            BayesianABDB.experiment_id == experiment_id,
-        )
-    )
+    conditions = [
+        BayesianABDB.workspace_id == workspace_id,
+        BayesianABDB.experiment_id == experiment_id,
+    ]
+
+    stmt = select(BayesianABDB).where(and_(*conditions))
     result = await asession.execute(stmt)
     return result.unique().scalar_one_or_none()
 
 
 async def delete_bayes_ab_experiment_by_id(
     experiment_id: int,
     user_id: int,
+    workspace_id: int,
     asession: AsyncSession,
 ) -> None:
     """
-    Delete the A/B experiment by id.
+    Delete the A/B experiment by id from a specific workspace.
     """
     stmt = delete(BayesianABDB).where(
         and_(
             BayesianABDB.user_id == user_id,
```

Comment:
Im confused why we still use `user_id` to filter. We should just be using `workspace_id` now right?